.github/workflows/go.yml

```diff
@@ -0,0 +1,49 @@
+name: Go
+
+on:
+  push:
+    branches:
+    - "main"
+  pull_request:
+    branches:
+    - "main"
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: [ "1.19", "1.20" ]
+    name: lint
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go }}
+      - run: go build -v ./...
+      - uses: Jerome1337/goimports-action@v1.0.2
+        with:
+          goimports-path: './'
+      - uses: iamnotaturtle/auto-gofmt@v2.1.0
+        with:
+          # This part is also where you can pass other options, for example:
+          only_changed: True
```

Comment:
not sure we need it, if we'd like to enforce some kind of formatting and imports sorting it would be nice to expose it to users outside of CI

Reply:
not quite sure what it means, these are here just to check if you ran them before submitting the code

Reply:
`iamnotaturtle/auto-gofmt@v2.1.0` IIRC uses gofumpt so it puts expectation on contributors to already have gofumpt installed and be aware of this CI step same with imports sorting, what I'm suggesting is if we are to keep these steps in CI perhaps it would be a good idea to prepare some make scripts so users can easily prepare their code for our CI or at least have this information in `CONTRIBUTING.md` or alike

Reply:
removed

Reply:
`CONTRIBUTING.md` - added

Reply:
I think this CONTRIBUTING.md needs some adjustments due to this being an OSS project, can we keep this PR simple, IMO it was fine without `auto-gofmt` and `goimports-action`, simple lint and tests is good for now, should we need more we can do this in separate PR with specific requirements, wdyt @rjeczalik